packages/plugin-express/src/express.js

```diff
@@ -29,24 +25,10 @@ module.exports = {
         const { metadata, request } = getRequestAndMetadataFromReq(req)
         event.request = { ...event.request, ...request }
         event.addMetadata('request', metadata)
+        event._handledState.severityReason = { type: 'unhandledErrorMiddleware' }
```

Comment:
I don't think this is quite right — if it's a handled error then it shouldn't be set to `unhandledErrorMiddleware` and we need the `attributes` set so the dashboard knows the error came from Express

There's a `handledState` variable already, so I think this needs to be:

```js
if (event.unhandled) {
  event.severity = 'error'
  event._handledState = handledState
}
```

Reply:
Hmm, forcing the severityReason here has opened up an issue for unhandled promise rejections. In node versions prior to 16 the AsyncLocalStorage context is not available in the unhandled rejection handler so you get an error with severityReason `unhandledPromiseRejection` and none of the additional context (e.g. request). However, for  node >= 16 the context is available so this new logic runs and sets `unhandledErrorMiddleware`. At this point I'm not sure which is more correct. Furthermore, presumably, we'd like this to not be dependent on the node version i.e. it's either always `unhandledPromiseRejection` or always `unhandledErrorMiddleware`.

Reply:
Right, I've addressed this so the behaviour is the same as before. But it did then raise the issue of errors coming from the unhandled promise rejection handler, which are of the following handled state:

```
{
  severity: 'error',
  unhandled: true,
  severityReason: {
    type: 'unhandledPromiseRejection',
  }
}
```

I think it's useful that we keep these as unhandled promise rejections as this type of error can be lacking in detail so knowing it was an unhandled promise rejection is useful information. So the total logic is now:

```
        if (event.unhandled) {
          const originalSeverityReasonType = event._handledState.severityReason.type        
          event.severity = 'error'
          event._handledState = handledState
          if (originalSeverityReasonType === 'unhandledPromiseRejection') {
            event._handledState.severityReason.type = 'unhandledPromiseRejection'
          }
        }
```

Which results in:

```
{
  severity: 'error',
  unhandled: true,
  severityReason: {
    type: 'unhandledPromiseRejection',
    attributes: { framework: 'Express/Connect' }
  }
}
```

Note that the framework attribute remains.

In the dashboard this type of error currently looks like:

<img width="300" alt="Screenshot 2023-02-23 at 14 11 17" src="https://user-images.githubusercontent.com/2514172/220931887-4e3dee84-6fb2-40d0-9c94-1c8ac9638be2.png">

With product steer I could make a change in the dashboard to also include the framework attribute, so that we surface the same data we do for an uncaught exception:

<img width="292" alt="Screenshot 2023-02-23 at 14 12 44" src="https://user-images.githubusercontent.com/2514172/220932238-ef444a44-baf7-40dc-9447-e9589f5b7d38.png">